roboflow/core/project.py

```diff
@@ -347,10 +347,13 @@ def __annotation_upload(self, annotation_path, image_id):
         # Get annotation string
         annotation_string = open(annotation_path, "r").read()
         # Set annotation upload url
+
+        project_name = self.id.rsplit("/")[1]   
+
         self.annotation_upload_url = "".join(
             [
                 API_URL + "/dataset/",
-                self.name,
+                project_name,
```

Comment:
this is what what causing the `Image not found in dataset.`. we saw sometimes. after having created a bunch of projects with the same name but different id/urls